spec/services/discovery_report_spec.rb

```diff
@@ -19,15 +19,24 @@
   end
 
   describe '#each_row' do
+    let(:dobj_hash1) { { druid: '1', counts: { total_size: 1, mimetypes: { a: 1, b: 2 } }, errors: {} } }
+    let(:dobj_hash2) { { druid: '2', counts: { total_size: 2, mimetypes: { b: 3, q: 4 } }, errors: {} } }
+    let(:dobj_hash3) { { druid: '3', counts: { total_size: 3, mimetypes: { q: 9 } }, errors: { foo: true } } }
+
```

Comment:
here you create fake data that did or didn't have errors.  but the discovery report is the code that is executed to put the errors in the hash.   So it feels correct to me for the discovery report spec to show that raw data A leads to error X in the hash.  The errors themselves are not created in digital_object code, nor in the templates.

Reply:
i would agree there should be a test of some sort that tests the aggregation of the info being mocked here, starting from fixture data.  doesn't seem like it has to happen in this PR, and this PR is addressing a specific ticket that doesn't include that in its scope.  but seems like a reasonable thing to ticket (and would be a test i don't think this codebase has ever had?).

Reply:
Previous discover_report code was entirely untested.  Not even "does this method exist?"

Reply:
see #298